### PR TITLE
update picmi user documentation

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request
- separates the PICMI python dependencies out of the global space and into a local `picmi/requirements.txt`
- updates the introduction to picmi to include what setup is needed to actually run the python example code

- [x] I need to check, whether the read the docs file is actually rendered correctly

### CODE REVIEW

File: lib/python/picongpu/picmi/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+typeguard >= 4.2.1
```
Comment (psychocoderHPC) [inline lib/python/picongpu/picmi/requirements.txt:1]: you must update https://github.com/ComputationalRadiationPhysics/picongpu/blob/800f7bc119d09a3d629aa34a80e2ff368d4289e5/share/ci/install/pypicongpu.sh#L43

to point to this file because the dependency generator can not handle the link in file `lib/python/picongpu/requirements.txt`

Comment (PrometheusPi) [reply lib/python/picongpu/picmi/requirements.txt:1]: Thank you @psychocoderHPC

File: docs/source/usage/picmi/intro.rst [hunk 2]
```diff
@@ -11,6 +11,17 @@ Example
...
+
+   Please be aware that for running the python script, you need to setup a python environment that includes all
+   dependecies listed in ``$PICSRC/lib/python/picongpu/picmi/requirements.txt``. This can be done by e.g.
+   ``pip install -r requirements.txt``. You also need to setup your ``$PYTHONPATH`` to include PIConGPU's
+   python libraries ``export PYTHONPATH=$PICSRC/lib/python:$PYTHONPATH``. (This is commonly done by your PIConGPU environment,
+   see below.)
+   For ``pic-build`` and ``tbg`` you need to setup a working PIConGPU environment. This is for example documented
+   in `the Setup part of PIConGPU in 5 Minutes on Hemera <hemeraIn5min>`. 
+
+
```
Comment (BrianMarre) [inline docs/source/usage/picmi/intro.rst]:
```suggestion
.. note::

Remember to install the PICMI python dependencies, see ``lib/python/picongpu/picmi/requirements.txt``, to your environment before using PICMI. This can be done by e.g. ``pip install -r requirements.txt`` (called from the PIConGPU source code directory).

In addition you must add the picongpu python modules to your PYTHONPATH, e.g. ``export PYTHONPATH=$PICSRC/lib/python:$PYTHONPATH`` and provide the environment variables,

- ``PICSRC`` ... path of picongpu source code directory
- ``PIC_EXAMPLES`` ... path to picongpu examples, usually ``$PICSRC/share/picongpu/examples`` 
- ``PIC_SYSTEM_TEMPLATE_PATH`` .. path to picongpu tpl file

For ``pic-build`` and ``tbg`` you need to setup a working PIConGPU environment. This is for example documented in `the Setup part of PIConGPU in 5 Minutes on Hemera <hemeraIn5min>`. 
```
In general this in formation should be moved to a separate installation documentation page and only referenced here. We suggest to do fix this in separate PR.

**Review by BrianMarre — COMMENTED** [review]
some suggestions

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -154,7 +154,7 @@ pypicongpu-generate-full-matrix:
     - apt install -y python3 python3-pip
     - pip3 install pyyaml requests
     - $CI_PROJECT_DIR/share/ci/git_merge.sh
-    - python3 $CI_PROJECT_DIR/share/ci/pypicongpu_generator.py $CI_PROJECT_DIR/lib/python/picongpu/requirements.txt > compile.yml
+    - python3 $CI_PROJECT_DIR/share/ci/pypicongpu_generator.py $CI_PROJECT_DIR/lib/python/picongpu/picmi/requirements.txt > compile.yml
```
Comment (PrometheusPi) [inline .gitlab-ci.yml:157]: This change is definitely needed

File: share/ci/install/pypicongpu.sh [hunk 4]
```diff
@@ -40,7 +40,7 @@ conda create -n pypicongpu python=${PYTHON_VERSION}
 conda activate pypicongpu
 python3 --version
 MODIFIED_REQUIREMENT_TXT=$CI_PROJECT_DIR/lib/python/picongpu/modified_requirements.txt
-python3 $CI_PROJECT_DIR/share/ci/install/requirements_txt_modifier.py $CI_PROJECT_DIR/lib/python/picongpu/requirements.txt $MODIFIED_REQUIREMENT_TXT
+python3 $CI_PROJECT_DIR/share/ci/install/requirements_txt_modifier.py $CI_PROJECT_DIR/lib/python/picongpu/picmi/requirements.txt $MODIFIED_REQUIREMENT_TXT
```
Comment (PrometheusPi) [inline share/ci/install/pypicongpu.sh:43]: I am not sure whether this is executed at some point of the CI

**Comment by PrometheusPi** [comment]
@BrianMarre I updated the text based on our discussion yesterday. Please take another stab at it.

**Review by BrianMarre — APPROVED** [review]
looks good

**Comment by PrometheusPi** [comment]
ping @psychocoderHPC @ikbuibui @steindev ready for review

File: share/ci/install/pypicongpu.sh (hunk 4 above)
Comment (ikbuibui) [reply share/ci/install/pypicongpu.sh:43]: It is tested in the `pypicongpu-full-matrix` test. The dependencies are installed.

**Review by ikbuibui — APPROVED** [review]
Looks good
